test/images/README.md

```diff
@@ -0,0 +1,117 @@
+# Kubernetes test images
+
+## Overview
+
+All the images found here are used in Kubernetes tests that ensure its features and functionality.
+The images are built and published as manifest lists, allowing multiarch and cross platform support.
+
+This guide will provide information on how to: make changes to images, bump their version, build the
+new images, test the changes made.
+
+
+## Prerequisites
+
+In order to build the docker test images, a Linux node is required. The node will require `make`
+and `docker (version 18.06.0 or newer)`. Manifest lists were introduced in 18.03.0, but 18.06.0
+is recommended in order to avoid certain issues.
+
+The node must be able to push the images to the desired container registry. Depending on the
+container registry, it might require a different authentication method. For dockerhub, this
+can be done by running the command:
+
+```bash
+    docker login -u your-awesome-username -p anAwesomerPassword
+```
+
+
+## Making changes to images
+
+There are several thousands of tests in Kubernetes E2E testing. Not all of them are being run on
+new PRs, and thus, not all images are used, especially those that are not used by Conformance tests.
+
+So, in order to prevent regressions in the images and failing jobs, any changes made to the image
+itself or its binaries will require the image's version to be bumped. In the case of a regression
+which cannot be immediately resolved, the image version used in E2E tests will be reverted to the
+last known stable version.
+
+The version can easily be bumped by modifying the file `test/images/IMAGE_NAME/VERSION`, which will
+be used when building the image. Additionally, for the `agnhost` image, also bump the `Version` in
+`test/images/agnhost/agnhost.go`.
+
+The typical image used in E2E testing is the `agnhost` image. It contains several subcommands with
+different [functionalities](agnhost/README.md), used to validate different Kubernetes behaviours. If
+a new functionality needs testing, consider adding an `agnhost` subcommand for it first, before
+creating an entirely separate test image.
+
+Some test images (`mounttest`, `test-webserver`) are used as bases for other images (`mounttest-user`,
+`kitten`, `nautilus`). If the parent image's `VERSION` has been bumped, also bump the version in the
+children's `BASEIMAGE` files in order for base image changes to be reflected in the child images as well.
+
+
+## Building images
+
+The images are built through `make`. Since some images (`mounttest`, `test-webserver`)
+are used as a base for other images, it is recommended to build them first, if needed.
+
+An image can be built by simply running the command:
+
+```bash
+    make all WHAT=test-webserver
```

Comment:
is there a reason we indented all of these code snippet contents? IIRC on some shells this will prevent them from showing up in history if naively copied?

Reply:
agree, if code is wrapped in triple-backticks, it doesn't need to be 4-space indented to pickup code formatting